### PR TITLE
kit chain

### PR DESCRIPTION
**Pull request by bitful-pannul**

### CODE REVIEW

File: src/kit/boot-fake-node.md [hunk 1]
```diff
@@ -9,21 +9,23 @@ kit boot-fake-node
 By default, `boot-fake-node` fetches a prebuilt binary and launches the node using it.
 Alternatively, `boot-fake-node` can use a local binary or build a local Kinode core repo and use the resulting binary.
 
+It also boots a fake chain with anvil in the background (see [kit chain](../kit/chain.md)). This comes preseeded with KNS & app_store contracts, that fake nodes use to index each others networking info.
```
Comment (nick1udwig) [inline src/kit/boot-fake-node.md:12]: 1. let's link `anvil` to some relevant foundry anvil docs page
2. one sentence per line, please add newline
3. let's reword the second sentence in two ways:

First, change the opening `This` to `The fake chain` for more specificity. Second, let's reword the second half of the sentence. We need to say what KNS is used for AND what app_store is used for and make it clear to which we are referring.

Comment (nick1udwig) [reply src/kit/boot-fake-node.md:12]: Possible rewording:
```
The fake chain comes preseeded with two contracts: KNS, which nodes use to index networking info of other nodes; and app_store, which _____
```

File: src/kit/boot-fake-node.md [hunk 2]
```diff
@@ -9,21 +9,23 @@ kit boot-fake-node
...
+
 ## Example Usage
 
 You can start a network of fake nodes that can communicate with each other (but not the live network).
 You'll need to start a new terminal for each fake node.
-For example, to start two fake nodes, `fake.os` and `fake2.os`:
+For example, to start two fake nodes, `fakedev` and `fake2.dev`:
```
Comment (nick1udwig) [inline src/kit/boot-fake-node.md:18]: Missing a period

File: src/kit/chain.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    0.2.0 (b51c057 2024-01-18T00:23:54.430340000Z)
+    https://github.com/foundry-rs/foundry
+```
+
+The default port is `8545` and the chain ID is `31337`.
+
+## Discussion
+
+`kit chain` starts an anvil node with the arguments `--load-state kinostate.json`. This json file includes the [KNS](https://github.com/kinode-dao/KNS) & app_store contracts, and is included in the kit binary.
```
Comment (nick1udwig) [inline src/kit/chain.md:25]: * One sentence per line
* kit should be `kit`

File: src/kit/chain.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+The default port is `8545` and the chain ID is `31337`.
+
+## Discussion
+
+`kit chain` starts an anvil node with the arguments `--load-state kinostate.json`. This json file includes the [KNS](https://github.com/kinode-dao/KNS) & app_store contracts, and is included in the kit binary. 
+
+The [kinostate.json](https://github.com/kinode-dao/kit/blob/master/src/chain/kinostate.json) file can be found written into /tmp/kinode-kit-cache/kinostate-{hash}.json upon running the command. 
+
+Note that while the kns_indexer and app_store apps in fake nodes use this chain to index events, any events loaded from a json statefile, aren't replayed upon restarting anvil. This is a future `foundry` feature request.
```
Comment (nick1udwig) [inline src/kit/chain.md:29]: One sentence per line. If it really is a feature request, plz open an issue

File: src/kit/chain.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+Usage: kit chain [OPTIONS]
+
+Options:
+  -p, --port <PORT>  Port to run the chain on [default: 8545]
+  -h, --help         Print help
+```
+
+### `--port`
+
+Port to run anvil fakechain on. Defaults to `8545`.
```
Comment (nick1udwig) [inline src/kit/chain.md:46]: One sentence per line

File: src/kit/chain.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+                             _   _
+                            (_) | |
+      __ _   _ __   __   __  _  | |
+     / _` | | '_ \  \ \ / / | | | |
+    | (_| | | | | |  \ V /  | | | |
+     \__,_| |_| |_|   \_/   |_| |_|
+
+    0.2.0 (b51c057 2024-01-18T00:23:54.430340000Z)
+    https://github.com/foundry-rs/foundry
+```
```
Comment (nick1udwig) [inline src/kit/chain.md:19]: We probably should get rid of this ascii art

**Review by nick1udwig — CHANGES_REQUESTED** [review]
Some comments; after @jurij-jukic s PRs we should also `grep -r` through the book for `.os` and change them all to `.dev`.

**Comment by bitful-pannul** [comment]
> Some comments; after @jurij-jukic s PRs we should also `grep -r` through the book for `.os` and change them all to `.dev`.

Addressed, thank you! Regarding changing `.os` to `.dev` everywhere, I defaulted in kit to .dev to show the possibility of contract development, here in the case of a custom TLD contract. 

But it could just as well use the .os one, and perhaps it would be clearer if it did?

**Review by nick1udwig — APPROVED** [review]
all -> `.dev`; otherwise LGTM

**Pull request merged**